bigframes/dataframe.py

```diff
@@ -3106,6 +3107,12 @@ def to_gbq(
                 + constants.DEFAULT_EXPIRATION,
             )
 
+        if len(labels) != 0:
+            client = self._session.bqclient
```

Comment:
Setting up labels currently requires additional client calls, which is acceptable but not ideal for performance. Could you try setting the label directly at the `bigquery.table.TableReference destination`? While `TableReference` doesn't have a public API for this, you can experiment with the internal call similar to [Table.label](https://github.com/googleapis/python-bigquery/blob/ea69fe315592f3a73417f175b1fe4543203cb716/google/cloud/bigquery/table.py#L480). If this works, we can explore adding a new public API to TableReference. +@tswast has more experience with the Python client API and could provide additional insights.

Reply:
I removed one client call by using the TableReference returned by the Query job, but as far as I can tell we might still have to make another API call for `update_table`. Looking at seeing if I can change table labels upon creation but I haven't found where that's being done yet.

Reply:
Thanks for exploring it.